bundles/org.openhab.binding.gardena/src/main/java/org/openhab/binding/gardena/internal/GardenaSmartWebSocket.java

```diff
@@ -139,18 +136,29 @@ public void onFrame(Frame pong) {
     @OnWebSocketClose
     public void onClose(int statusCode, String reason) {
         if (!closing) {
-            logger.debug("Connection to Gardena Webservice was closed ({}): code: {}, reason: {}", socketId, statusCode,
+            logger.warn("Connection to Gardena Webservice was closed ({}): code: {}, reason: {}", socketId, statusCode,
                     reason);
-            socketEventListener.onWebSocketClose();
+
+            // let listener handle restart of socket
+            socketEventListener.onWebSocketClose(locationID);
         }
     }
 
     @OnWebSocketError
     public void onError(Throwable cause) {
         if (!closing) {
-            logger.warn("Gardena Webservice error ({}): {}, restarting", socketId, cause.getMessage());
-            logger.debug("{}", cause.getMessage(), cause);
-            socketEventListener.onWebSocketError();
+            if (logger.isDebugEnabled()) {
+                logger.warn("Gardena Webservice error ({})", socketId, cause);
```

Comment:
And (see above) again, why logger.warn()?

Reply:
I use logger.warn, because there was some kind of error other than an expired token. (I did not change the logging level here. It already was set to "warn" before my changes.)

But I would be fine with using logger.info here as well. 
So, this would result in:
Connection closed (Expired token) -> `log.debug`
Connection closed (other reason) -> `log.info`
Connection error -> `log.info`

I also would like to move the logging in `onClose` and `onError` outside of the `if (!closing) {...}` block to always log everything in case `onClose` and `onError` occurs simultaneously

Reply:
Makes sense.